examples/cookbook/agents/documentation_agent/agent.py

```diff
@@ -0,0 +1,170 @@
+    retriever = VectorIndexRetriever(
+        index=cast(VectorStoreIndex, loaded_index),
+        similarity_top_k=10,
+    )
+    retrieved_nodes = retriever.retrieve(query_bundle)
+    choice_batch_size = 5
+    top_n = 2
+    results: list[Relevance] = []
+    for idx in range(0, len(retrieved_nodes), choice_batch_size):
+        nodes_batch = [
+            {
+                "id": idx + id,
+                "text": node.node.get_text(),  # pyright: ignore[reportAttributeAccessIssue]
+                "document_title": node.metadata["document_title"],
+                "semantic_score": node.score,
+            }
+            for id, node in enumerate(retrieved_nodes[idx : idx + choice_batch_size])
+        ]
+        results += llm_query_rerank(nodes_batch, query)
+    results = sorted(results, key=lambda x: x.score or 0, reverse=True)[:top_n]
+
+    return [result.document for result in results]
+
+
+class Response(BaseModel):
+    classification: Literal["code", "general"] = Field(
+        ..., description="The classification of the question"
+    )
+    content: str = Field(..., description="The response content")
+
+
+class DocumentationAgent(BaseModel):
+    @openai.call("gpt-4o-mini", response_model=Response, json_mode=True)
+    @prompt_template(
+        """
+        SYSTEM:
+        You are an AI Assistant that is an expert at answering questions about Mirascope.
+        Here is the relevant documentation to answer the question.
+
+        First classify the question into one of two types:
+            - General Information: Questions about the system or its components.
+            - Code Examples: Questions that require code snippets or examples.
+
+        For General Information, provide a summary of the relevant documents 
+        if the question is too broad ask for more details. If the context does not
+        answer the question, say that the information is not available 
+        or you could not find it.
+
+        For Code Examples, output ONLY code without any markdown, with comments if necessary.
+        If the context does not answer the question, 
+        say that the information is not available.
+
+        Examples:
+            Question: "What is Mirascope?"
+            Answer:
+            A toolkit for building AI-powered applications with Large Language Models (LLMs).
+
+            Explanation: This is a General Information question, so a summary is provided.
+
+            Question: "How do I make a basic OpenAI call using Mirascope?"
+            Answer:
+            from mirascope.core import openai, prompt_template
+
+
+            @openai.call("gpt-4o-mini")
+            @prompt_template("Recommend a {{genre}} book")
+            def recommend_book(genre: str):
+                ...
+
+            response = recommend_book("fantasy")
+            print(response.content)
+
+            Explanation: This is a Code Examples question, 
+            so only a code snippet is provided.
```

Comment:
any reason this is on a newline?

Reply:
this was resolved but I'm still seeing it on a newline with no explanation, so either the change wasn't pushed or there's a reason there's a newline?

Reply:
oh oops, there were two so i only changed one